polkadot/node/service/src/overseer.rs

```diff
@@ -508,15 +487,16 @@ pub trait OverseerGen {
 	fn generate<Spawner, RuntimeClient>(
 		&self,
 		connector: OverseerConnector,
-		args: OverseerGenArgs<Spawner, RuntimeClient>,
+		args: OverseerGenArgs<Spawner, DefaultSubsystemClient<RuntimeClient>>,
```

Comment:
```suggestion
		args: OverseerGenArgs<Spawner, RuntimeClient>,
```

Nit: I think one last simplification that would make sense to me is to change this to just `RuntimeClient` too. The return type then also needs to be `Arc<RuntimeClient>`.

[Here](https://github.com/paritytech/polkadot-sdk/blob/165d075a5f680791c8e0b128ea60edd590a44f1f/polkadot/node/service/src/lib.rs#L1086) we would need to adjust to `.generate::<service::SpawnTaskHandle, DefaultSubsystemClient<FullClient>>`, but thats fine.

Reply:
refactored at https://github.com/paritytech/polkadot-sdk/pull/3335/commits/21224d98da51509216a278aa4e22aac420db3c7b